### ISSUE TITLE
F2 Score and mAP metric for object detection model.

### ISSUE THREAD

**Issue by Suzan009** [tensorflow/tensorflow#53487]
I didn't find a metric regarding the object detection task in keras. 

How to implement the above two metrics in keras and use them? In this competition, https://www.kaggle.com/c/tensorflow-great-barrier-reef/overview/evaluation uses the F2 score for its object detection task but I don't know how to evaluate my model locally with this score? 

> The metric sweeps over IoU thresholds in the range of 0.3 to 0.8 with a step size of 0.05, calculating an F2 score at each threshold. For example, at a threshold of 0.5, a predicted object is considered a "hit" if its IoU with a ground truth object is at least 0.5.

**Comment by sushreebarsa** [comment]
@Suzan009 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thanks!

**Comment by Suzan009** [comment]
I did. But I also want to achieve this in pure tensorflow.

**Comment by Saduf2019** [comment]
@Suzan009 
Please follow the TFOD API code and tensorflow garden, as this repo is for bugs and feature request, we request you to move this issue to closed status and for any further queries open a new issue at discussion forum which has larger community to help.
Thanks!

**Comment by Suzan009** [comment]
@Saduf2019 I am not following the messy repo of the [tensorflow-model garden](https://github.com/tensorflow/models/tree/master/research/object_detection). In fact, this so-called model garden repo should be closed. 

I'm simply looking to get the implementation of these two metrics in pure tensorflow. Asking or suggesting to look into TFOD API, you're making someone's life harder.

**Comment by Saduf2019** [comment]
@Suzan009 

You may take the following command as a reference to get the mAP values.

python models/research/object_detection/model_main_tf2.py 
--model_dir = <YOUR_MODEL_PATH>
--pipeline_config_path = <YOUR_PIPELINE.CONFIG_PATH>
--checkpoint_dir = <YOUR_CHECKPOINT_DIR_PATH>

Alternatively, you can visualize these metrics on Tensorboard by initiating Tensorboard in the eval directory.
As informed earlier this repo is for bugs you have to open this issue in tf discussion forum.
